log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/ThrowableRenderer.java

```diff
@@ -0,0 +1,290 @@
+            renderThrowable(
+                    buffer, suppressedThrowable, context, visitedThrowables, prefix, lineSeparator, SUPPRESSED_CAPTION);
+        }
+    }
+
+    void renderCause(
+            final StringBuilder buffer,
+            final Throwable cause,
+            final C context,
+            final Set<Throwable> visitedThrowables,
+            final String prefix,
+            final String lineSeparator) {
+        if (cause != null) {
+            renderThrowable(buffer, cause, context, visitedThrowables, prefix, lineSeparator, CAUSED_BY_CAPTION);
+        }
+    }
+
+    static void renderThrowableMessage(final StringBuilder buffer, final Throwable throwable) {
+        final String message = throwable.getMessage();
+        buffer.append(throwable.getClass().getName());
+        if (message != null) {
+            buffer.append(": ");
+            buffer.append(message);
+        }
+    }
+
+    final void renderStackTraceElements(
+            final StringBuilder buffer,
+            final Throwable throwable,
+            final C context,
+            final String prefix,
+            final String lineSeparator) {
+        context.ignoredStackTraceElementCount = 0;
+        final Context.Metadata metadata = context.metadataByThrowable.get(throwable);
+        final StackTraceElement[] stackTraceElements = throwable.getStackTrace();
+        for (int i = 0; i < metadata.stackLength; i++) {
+            renderStackTraceElement(buffer, stackTraceElements[i], context, prefix, lineSeparator);
+        }
+        if (context.ignoredStackTraceElementCount > 0) {
+            renderSuppressedCount(buffer, context.ignoredStackTraceElementCount, prefix, lineSeparator);
+        }
+        if (metadata.commonElementCount != 0) {
+            buffer.append(prefix);
+            buffer.append("\t... ");
+            buffer.append(metadata.commonElementCount);
+            buffer.append(" more");
+            buffer.append(lineSeparator);
+        }
+    }
+
+    void renderStackTraceElement(
+            final StringBuilder buffer,
+            final StackTraceElement stackTraceElement,
+            final C context,
+            final String prefix,
+            final String lineSeparator) {
+
+        // Short-circuit on ignored stack trace elements
+        final boolean stackTraceElementIgnored = isStackTraceElementIgnored(stackTraceElement, ignoredPackageNames);
+        if (stackTraceElementIgnored) {
+            context.ignoredStackTraceElementCount += 1;
+            return;
+        }
+
+        // Render the stack trace element
+        if (context.ignoredStackTraceElementCount > 0) {
+            renderSuppressedCount(buffer, context.ignoredStackTraceElementCount, prefix, lineSeparator);
+            context.ignoredStackTraceElementCount = 0;
+        }
+        buffer.append(prefix);
+        buffer.append("\tat ");
+        buffer.append(stackTraceElement.toString());
+        buffer.append(lineSeparator);
+    }
```

Comment:
Where are my colors, I want my colors back! :wink:

We probably need to use `TextRenderer` here, which allows to intermingle the text with ANSI escapes.

Reply:
I prefer not to support/use `TextRenderer` anymore; it is an incomplete implementation, it is non-vital, and it will be dropped in Log4j 3.

Are you strongly objecting? Can we convince you the other way around?

Reply:
You don't need a lot to convince me. :wink:

You can drop it, but please be prepared to reimplement it if a user complains. Since you are dropping it, please drop `TextRenderer` support from `MessagePatternConverter` too.